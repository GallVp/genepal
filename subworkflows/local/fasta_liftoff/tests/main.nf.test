nextflow_workflow {

    name "Test Subworkflow FASTA_LIFTOFF"
    script "../main.nf"
    workflow "FASTA_LIFTOFF"
    config './nextflow.config'

    tag "subworkflows"
    tag "subworkflows_gallvp"
    tag "subworkflows/fasta_liftoff"
    tag "subworkflows/gff_tsebra_spfilterfeaturefromkilllist"

    tag "gunzip"
    tag "gffread"
    tag "liftoff"
    tag "agat"
    tag "agat/spmergeannotations"
    tag "agat/spflagshortintrons"
    tag "agat/spfilterfeaturefromkilllist"

    setup {
        run('GUNZIP', alias: 'GUNZIP_GENOME_FASTA') {
            script "../../../../modules/nf-core/gunzip"

            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file(params.modules_testdata_base_path + 'genomics/eukaryotes/actinidia_chinensis/genome/chr1/genome.fasta.gz', checkIfExists: true)
                ]
                """
            }
        }

        run('GUNZIP', alias: 'GUNZIP_BRAKER_HINTS') {
            script "../../../../modules/nf-core/gunzip"

            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file(params.modules_testdata_base_path + 'genomics/eukaryotes/actinidia_chinensis/genome/chr1/genome.hints.gff.gz', checkIfExists: true)
                ]
                """
            }
        }
    }


    test("liftoff - GCF_019202715 - to - actinidia_chinensis") {

        when {
            workflow {
                """
                input[0]    = GUNZIP_GENOME_FASTA.out.gunzip

                input[1]    = Channel.of([
                    [ id:'ref' ],
                    file ( "${baseDir}/subworkflows/local/fasta_liftoff/tests/testdata/GCF_019202715.1.fna.gz", checkIfExists: true )
                ])
                input[2]    = Channel.of([
                    [ id:'ref' ],
                    file ( "${baseDir}/subworkflows/local/fasta_liftoff/tests/testdata/GCF_019202715.1.gff.gz", checkIfExists: true )
                ])

                input[3]    = true // val_filter_liftoff_by_hints

                input[4]    = GUNZIP_BRAKER_HINTS.out.gunzip

                input[5]    = Channel.of ( file("${baseDir}/assets/tsebra-template.cfg", checkIfExists: true) )
                            | map { cfg ->
                                def enforce_full_intron_support = true
                                def param_intron_support = enforce_full_intron_support ? '1.0' : '0.0'

                                def param_e1 = params.allow_isoforms ? '0.1'    : '0.0'
                                def param_e2 = params.allow_isoforms ? '0.5'    : '0.0'
                                def param_e3 = params.allow_isoforms ? '0.05'   : '0.0'
                                def param_e4 = params.allow_isoforms ? '0.2'    : '0.0'

                                [
                                    'tsebra-config.cfg',
                                    cfg
                                    .text
                                    .replace('PARAM_INTRON_SUPPORT', param_intron_support)
                                    .replace('PARAM_E1', param_e1)
                                    .replace('PARAM_E2', param_e2)
                                    .replace('PARAM_E3', param_e3)
                                    .replace('PARAM_E4', param_e4)
                                ]
                            }
                            | collectFile

                input[6]    = false // val_allow_isoforms
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }
}
