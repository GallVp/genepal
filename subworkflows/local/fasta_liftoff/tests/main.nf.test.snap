{
    "liftoff - GCF_019202715 - to - actinidia_chinensis": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gff:md5,b2b6a54f44df1dc2a324e7c448f8703c"
                    ]
                ],
                "1": [
                    "versions.yml:md5,19d093420a74661d71e3690828949796",
                    "versions.yml:md5,1b4c11d661d6f5415fba4c381299308b",
                    "versions.yml:md5,1cafcdb9de6b733da7979604133eae25",
                    "versions.yml:md5,3fa77145c369d5322ea24769c9222947",
                    "versions.yml:md5,441fcf0a52433b408072a27ece6ae016",
                    "versions.yml:md5,4cee921a63318a115b0b4ea63545efc4",
                    "versions.yml:md5,7a863df1624e9daeb1065a18723a7181",
                    "versions.yml:md5,7ce16ccb5b6a919500779dac3a3476cc",
                    "versions.yml:md5,90f389011e267060373c0e2810d6ca0f",
                    "versions.yml:md5,c1f60e3da5457e968b8689d5e43ebdac",
                    "versions.yml:md5,ee651692163b338557acdd094a5d2bb8"
                ],
                "gff3": [
                    [
                        {
                            "id": "test"
                        },
                        "test.gff:md5,b2b6a54f44df1dc2a324e7c448f8703c"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,19d093420a74661d71e3690828949796",
                    "versions.yml:md5,1b4c11d661d6f5415fba4c381299308b",
                    "versions.yml:md5,1cafcdb9de6b733da7979604133eae25",
                    "versions.yml:md5,3fa77145c369d5322ea24769c9222947",
                    "versions.yml:md5,441fcf0a52433b408072a27ece6ae016",
                    "versions.yml:md5,4cee921a63318a115b0b4ea63545efc4",
                    "versions.yml:md5,7a863df1624e9daeb1065a18723a7181",
                    "versions.yml:md5,7ce16ccb5b6a919500779dac3a3476cc",
                    "versions.yml:md5,90f389011e267060373c0e2810d6ca0f",
                    "versions.yml:md5,c1f60e3da5457e968b8689d5e43ebdac",
                    "versions.yml:md5,ee651692163b338557acdd094a5d2bb8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-12-05T12:49:44.433235"
    }
}